meshnets/train.py

```diff
@@ -91,25 +93,29 @@ def main(_):
         'latent_size': FLAGS.latent_size,
         'num_mlp_layers': FLAGS.num_mlp_layers,
         'message_passing_steps': FLAGS.message_passing_steps,
-        'learning_rate': FLAGS.learning_rate
-    }
-    experiment_config = {
+        'learning_rate': FLAGS.learning_rate,
         'experiment_name': FLAGS.experiment_name,
         'num_workers_loader': FLAGS.num_workers_loader,
-        'tuning_run': TUNING_RUN,
-        'num_workers_ray': FLAGS.num_workers_ray,
-        'num_cpus_per_worker': FLAGS.num_cpus_per_worker,
-        'use_gpu': FLAGS.use_gpu,
         'max_epochs': FLAGS.max_epochs,
         'log_every_n_steps': FLAGS.log_every_n_steps,
         'save_top_k': FLAGS.save_top_k,
+        'train_dataset': train_dataset,
+        'validation_datasets_names': val_datasets_names,
+        'validation_datasets': val_datasets
     }
 
-    model_training.train_model(config,
-                               experiment_config,
-                               train_dataset,
-                               validation_datasets_names=val_datasets_names,
-                               validation_datasets=val_datasets)
+    resources_per_worker = {
+        'CPU': FLAGS.num_cpus_per_worker,
+        'GPU': 1 if FLAGS.use_gpu else 0
+    }
+    scaling_config = ScalingConfig(num_workers=FLAGS.num_workers_ray,
+                                   use_gpu=FLAGS.use_gpu,
```

Comment:
What is the difference of this part and the one passing the GPU in the resources_per_worker?

Reply:
This chooses the resources for the function. The cluster is configured outside the script.

We can submit this to the cluster and then it will look for workers capable of meeting the demands.